### PR TITLE
SRV-100 Add ad payments fees (supply)

### PR DESCRIPTION
**Pull request by PawelPodkalicki**

### CODE REVIEW

**Comment by PawelPodkalicki** [comment]
I will refactor PaymentDetailsProcessor

File: app/Client/Mapper/AdSelect/EventPaymentMapper.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ class EventPaymentMapper
     public static function map(array $event): array
     {
         $mappedEvent = EventMapper::map($event);
-        $mappedEvent['paid_amount'] = $event['paid_amount'];
+        $mappedEvent['paid_amount'] = $event['event_value']; // AdSelect should have value before fees subtraction
```
Comment (teghnet) [inline app/Client/Mapper/AdSelect/EventPaymentMapper.php]: Do we need this comment?

File: app/Console/Commands/AdsProcessTx.php [hunk 2]
```diff
@@ -209,10 +212,17 @@ private function handleIfEventPayment(AdsPayment $dbTx): bool
...
         try {
-            $this->paymentDetailsProcessor->processPaymentDetails($senderAddress, $paymentId, $paymentDetails);
-        } catch (InvalidPaymentDetailsException $exception) {
-            // TODO log that demand send invalid payment
-            return false;
+            $this->paymentDetailsProcessor->processPaymentDetails($senderAddress, $adsPaymentId, $paymentDetails);
+        } catch (MissingInitialConfigurationException $exception) {
+            $this->error('Missing initial configuration: '.$exception->getMessage());
+
+            // transaction will be processed again later
```
Comment (teghnet) [inline app/Console/Commands/AdsProcessTx.php]: remove

File: app/Console/Commands/AdsProcessTx.php [hunk 3]
```diff
@@ -209,10 +212,17 @@ private function handleIfEventPayment(AdsPayment $dbTx): bool
...
+            return true;
+        } catch (Exception $exception) {
+            $this->error('Unexpected error: '.$exception->getMessage());
+
+            // transaction will be processed again later
```
Comment (teghnet) [inline app/Console/Commands/AdsProcessTx.php]: remove

**Review by teghnet — APPROVED** [review]

**Pull request merged**